src/base/http/responsegenerator.cpp

```diff
@@ -33,12 +33,32 @@
 
 #include "base/utils/gzip.h"
 
+namespace
+{
+    QString getCachingInterval(QString contentType)
+    {
+        contentType = contentType.toLower();
+
+        if (contentType.startsWith(QLatin1String("image/")))
+            return QLatin1String("private, max-age=604800");
+
+        if ((contentType == Http::CONTENT_TYPE_CSS)
+            || (contentType == Http::CONTENT_TYPE_JS)) {
+            // short interval in case of program update
+            return QLatin1String("private, max-age=43200");
+        }
+
+        return QLatin1String("no-store");
+    }
+}
+
 QByteArray Http::toByteArray(Response response)
 {
     compressContent(response);
 
     response.headers[HEADER_CONTENT_LENGTH] = QString::number(response.content.length());
     response.headers[HEADER_DATE] = httpDate();
+    response.headers[HEADER_CACHE_CONTROL] = getCachingInterval(response.headers.value(HEADER_CONTENT_TYPE));
```

Comment:
Are you sure it's universal behavior for all use cases? I mean embedded Tracker.

Reply:
I didn't noticed that part, I moved the implementation.